addons/im_livechat/controllers/main.py

```diff
@@ -69,13 +69,20 @@ def livechat_init(self, channel_id):
 
     @http.route('/im_livechat/get_session', type="json", auth='public')
     def get_session(self, channel_id, anonymous_name, **kwargs):
-        # if geoip, add the country name to the anonymous name
-        if request.session.geoip:
-            anonymous_name = anonymous_name + " ("+request.session.geoip.get('country_name', "")+")"
+        user_id = None
+        country_id = None
         # if the user is identifiy (eg: portal user on the frontend), don't use the anonymous name. The user will be added to session.
         if request.session.uid:
-            anonymous_name = request.env.user.name
-        return request.env["im_livechat.channel"].with_context(lang=False).get_mail_channel(channel_id, anonymous_name)
+            user_id = request.env.user.id
+            country_id = request.env.user.country_id.id
+        # if geoip, add the country name to the anonymous name
+        elif request.session.geoip:
+            # get the country of the anonymous person, if any
+            country_code = request.session.geoip.get('country_code', "")
+            country = request.env['res.country'].sudo().search([('code', '=', country_code)], limit=1) if country_code else None
+            if country:
+                anonymous_name, country_id = _("%s (%s)") % (anonymous_name, country.name), country.id
```

Comment:
Carefull here: you put the id of the country instead of its code. So my channel is named "Visitor, Mitchell Admin (2)" :s